src/app/member/components/MemberEditPage.test.tsx

```diff
@@ -0,0 +1,112 @@
+import {
+  render,
+  screen,
+  fireEvent,
+  waitFor,
+  act,
+  findByText,
+} from "@testing-library/react";
+import { beforeEach, describe, expect, it, vi } from "vitest";
+import { MemberEditPage } from "./MemberEditPage";
+import { onSubmit } from "./memberActions";
+import { Duties, User, UserRole } from "@/constant/constant";
+import userEvent from "@testing-library/user-event";
+import { format } from "date-fns";
+// vi.mock("../../../actions/memberActions", () => ({
+//   onSubmit: vi.fn(),
+// }));
+class MockPointerEvent extends Event {
+  button: number;
+  ctrlKey: boolean;
+  pointerType: string;
+
+  constructor(type: string, props: PointerEventInit) {
+    super(type, props);
+    this.button = props.button || 0;
+    this.ctrlKey = props.ctrlKey || false;
+    this.pointerType = props.pointerType || "mouse";
+  }
+}
+window.PointerEvent = MockPointerEvent as any;
+window.HTMLElement.prototype.scrollIntoView = vi.fn();
+window.HTMLElement.prototype.releasePointerCapture = vi.fn();
+window.HTMLElement.prototype.hasPointerCapture = vi.fn();
+type TestElement = Document | Element | Window | Node;
+
+// function for testing input values
+function hasInputValue(e: TestElement, inputValue: string) {
+  console.log("hasInputValue", e, inputValue);
+  return screen.getByDisplayValue(inputValue) === e;
+}
+// Mock Function for testing
+vi.mock("./memberActions.ts", { spy: true });
+vi.mock("date-fns", { spy: true });
+
+// Unit testing
+describe("UserForm", () => {
+  it("validates the form fields", async () => {
+    render(<MemberEditPage />);
+    // Inputs
+    const input = screen.getByTestId("name");
+    fireEvent.change(input, { target: { value: "123" } });
+    expect(hasInputValue(input, "123")).toBe(true);
```

Comment:
Let's create an array of targets and values and get rid of redundancy here:
```ts
const formTargets = [
{
testId: 'name',
value: '123'
}
]
```

Reply:
fixed